### ISSUE TITLE
Support Reusing Preconditioner for Adjoint

### ISSUE THREAD

**Issue by JaredCrean2** [OptimalDesignLab/PDESolver.jl#111]
If we use an approximate factorization preconditioner for solving a steady Newton iteration, we should find a way to reuse it for solving the adjoint.

**Comment by jehicken** [comment]
Yes, the last iteration of Newton may be close enough that the already-factored preconditioner will work well for the adjoint.
